store/src/test/java/com/dropbox/android/external/store4/util/InMemoryPersister.kt

```diff
@@ -0,0 +1,38 @@
+package com.dropbox.android.external.store4.util
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+/**
+ * An in-memory non-flowing persister for testing.
+ */
+class InMemoryPersister<Key, Output> {
+    private val data = mutableMapOf<Key, Output>()
+
+    @Suppress("RedundantSuspendModifier") // for function reference
+    suspend fun read(key: Key) = data[key]
+
+    @Suppress("RedundantSuspendModifier") // for function reference
+    suspend fun write(key: Key, output: Output) {
+        data[key] = output
+    }
+
+    @Suppress("RedundantSuspendModifier") // for function reference
+    suspend fun deleteByKey(key: Key) {
+        data.remove(key)
+    }
+
+    @Suppress("RedundantSuspendModifier") // for function reference
+    suspend fun deleteAll() {
+        data.clear()
+    }
+
+    fun peekEntry(key: Key): Output? {
+        return data[key]
+    }
+}
+
+@ExperimentalCoroutinesApi
+suspend fun <Key, Output> InMemoryPersister<Key, Output>.asObservable() = SimplePersisterAsFlowable(
+    reader = this::read,
+    writer = this::write
```

Comment:
don't you need to pass `delete` and `deleteAll` for you delete tests?

Reply:
`SimplePersisterAsFlowable` is only used by `FlowStoreTest`. My tests only use `InMemoryPersister`.

I'll move this extension to `SimplePersisterAsFlowable.kt` to make it clear.